src/main/java/com/rackspace/salus/telemetry/repositories/ResourceRepository.java

```diff
@@ -37,4 +39,7 @@
   Optional<Resource> findByTenantIdAndResourceId(String tenantId, String resourceId);
 
   List<Resource> findAllByTenantIdAndPresenceMonitoringEnabled(String tenantId, boolean presenceMonitoringEnabled);
+
+  @Query("from Resource r where r.id in :resourceIds")
+  Page<Resource> findByResourceId(List<Long> resourceIds, Pageable page);
```

Comment:
I couldn't find [where in the docs](https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#repositories.query-methods.details) this is described, but the phrase "In" in the method name allows for the equivalent of what you handcrafted in the query. IntelliJ is really good about auto-suggesting the JPA query method options:

![image](https://user-images.githubusercontent.com/988985/80230012-e950a380-8616-11ea-9649-50aaad4b111b.png)
```suggestion
  Page<Resource> findByResourceIdIn(List<Long> resourceIds, Pageable page);
```

Reply:
So this is what took me so long to try and find because I was fairly certain it had to exist in the magic function name header but like you said I couldn't find it in the documentation.